models/cart-discount/src/cart-discount-draft/generator.ts

```diff
@@ -0,0 +1,42 @@
+import * as CartDiscountValueAbsolute from '@commercetools-test-data/cart-discount-value-absolute';
+import * as CartDiscountValueFixed from '@commercetools-test-data/cart-discount-value-fixed';
+import * as CartDiscountValueGiftLineItem from '@commercetools-test-data/cart-discount-value-gift-line-item';
+import * as CartDiscountValueRelative from '@commercetools-test-data/cart-discount-value-relative';
+import { LocalizedString } from '@commercetools-test-data/commons';
+import { fake, Generator } from '@commercetools-test-data/core';
+import { createRelatedDates } from '@commercetools-test-data/utils';
+import { TCartDiscountDraft } from '../types';
+
+// https://docs.commercetools.com/api/projects/cartDiscounts#cartdiscountdraft
+
+// eslint-disable-next-line
+const [getValidFrom, _, getValidUntil] = createRelatedDates();
+
+const generator = Generator<TCartDiscountDraft>({
+  fields: {
+    key: fake((f) => f.lorem.slug(2)),
+    name: fake(() => LocalizedString.random()),
+    description: fake(() => LocalizedString.random()),
+    value: fake((f) =>
+      f.helpers.arrayElement([
+        CartDiscountValueAbsolute.CartDiscountValueAbsoluteDraft.random(),
+        CartDiscountValueFixed.CartDiscountValueFixedDraft.random(),
+        CartDiscountValueGiftLineItem.CartDiscountValueGiftLineItemDraft.random(),
+        CartDiscountValueRelative.CartDiscountValueRelativeDraft.random(),
+      ])
+    ),
+    cartPredicate: '1=1',
+    target: null,
+    sortOrder: fake((f) => String(Math.random())),
```

Comment:
This needs to be between 0 and 1

Reply:
Thanks! dc3df4853128bf7da52f0700bf3d6a42273ac13a should do it